File: 2023/UICPC Regional/Round #12/solutions/stockbroker/per.cc
#include <bits/stdc++.h>
typedef long long ll;

int main(void) {
  ll money = 100, days, prev = 1<<30, cur;
  std::cin >> days;
  for (int i = 0; i < days; ++i, prev=cur) {
    std::cin >> cur;
    if (cur > prev)
      money += std::min(money/prev, 100000LL)*(cur-prev);
  }
  std::cout << money << std::endl;
  return 0;
}

File: 2023/UICPC Regional/Round #12/solutions/stockbroker/per.php
<?php

fscanf(STDIN, "%d", $days);
$money = 100;
$prev = 1<<30;
for ($i = 0; $i < $days; ++$i) {
    fscanf(STDIN, "%d", $cur);
    if ($cur > $prev)
       $money += min(floor($money/$prev), 100000)*($cur-$prev);
    $prev = $cur;       	      	 		   
}
echo $money;
